pypiserver/_app.py

```diff
@@ -59,6 +59,13 @@ def log_request():
     log.info(config.log_req_frmt, request.environ)
 
 
+@app.hook('before_request')
+def print_request():
+    parsed = urlparse(request.urlparts.scheme + "://" + request.urlparts.netloc)
+    request.custom_host = parsed.netloc
+    request.custom_fullpath = parsed.path.rstrip('/') + '/' + request.fullpath.lstrip('/')
```

Comment:
I used the names `request.custom_<property>` for additional properties to explicitly specify that these are not from `bottle.py`.